libraries/AP_InertialSensor/AP_InertialSensor_MPU9250.cpp

```diff
@@ -421,8 +417,14 @@ bool AP_InertialSensor_MPU9250::_hardware_init(void)
     // Chip reset
     uint8_t tries;
     for (tries = 0; tries<5; tries++) {
+        // reset device
+        _register_write(MPUREG_PWR_MGMT_1, BIT_PWR_MGMT_1_DEVICE_RESET);
+
         hal.scheduler->delay(100);
 
+        // disable I2C as recommended by the datasheet
+        _register_write(MPUREG_USER_CTRL, BIT_USER_CTRL_I2C_IF_DIS);
```

Comment:
have you checked whether this impacts the use of the in-built AK8963 mag? (I seem to remember we had to have I2C enabled for that)